src/mappings/entities/mapPortfolio.ts

```diff
@@ -39,6 +40,30 @@ export const createPortfolio = (
   }).save();
 };
 
+/**
+ * Creates a Portfolio if not present.
+ * This can be need in case where someone creates an Instruction before the recipient Portfolio is created
+ */
+export const createPortfolioIfNotExists = async (
+  { identityId, number }: Pick<Portfolio, 'identityId' | 'number'>,
+  blockId: string,
+  event: SubstrateEvent
+): Promise<void> => {
+  await createIdentityIfNotExists(identityId, blockId, event);
```

Comment:
What about inlining this function so people don't use it by mistake? We'd be able to batch some requests as well.

Reply:
(might be better to leave as is though)

Reply:
Inlining would mean to handle `Identity` entity changes here which I have tried to avoid. My idea was to keep the respective entity operations in its respective mapping handler (like all `Portfolio.get`, `Portfolio.create`, `Portfolio.remove` operations would be in `mapPortfolios` and we would have exported members if something is needed elsewhere)